mars/services/task/execution/ray/executor.py

```diff
@@ -368,6 +371,74 @@ class _RayMonitorContext:
     )
 
 
+@dataclass
+class _RaySubtaskRuntime:
+    start_time: float = 0.0
+
+
+class _RaySlowSubtaskChecker:
+    @dataclass
+    class _CheckInfo:
+        count: int
+        upper: float
+
+    def __init__(
+        self,
+        total_subtask_count: int,
+        submitted_subtasks: OrderedSet,
+        completed_subtasks: OrderedSet,
+    ):
+        self._total_subtask_count = total_subtask_count
+        self._submitted_subtasks = submitted_subtasks
+        self._completed_subtasks = completed_subtasks
+        self._logic_key_to_subtask_costs = collections.defaultdict(list)
```

Comment:
homogeneous subtasks relies on symmetric subtask graph, would be better to be consistent with existing slow task detection logic

Reply:
This is similar with existing slow task detection logic, the subtask durations are grouped by the logic key.